packages/typespec-ts/src/modular/buildProjectFiles.ts

```diff
@@ -343,6 +348,152 @@ export function emitPackage(
+      "lint:fix": "echo skipped",
+      lint: "echo skipped",
+      pack: "npm pack 2>&1",
+      test: 'echo "Error: no test specified" && exit 1',
+      build:
+        "npm run clean && tsc && rollup -c 2>&1 && npm run minify && mkdirp ./review && npm run extract-api",
+      minify:
+        "uglifyjs -c -m --comments --source-map \"content='./dist/index.js.map'\" -o ./dist/index.min.js ./dist/index.js"
+    },
+    sideEffects: false,
+    autoPublish: false,
+    dependencies: {
+      tslib: "^2.2.0",
+      "@typespec/ts-http-runtime": runtimeLibVersion
+    },
+    devDependencies: {
+      "@microsoft/api-extractor": "^7.31.1",
+      "@types/node": "^18.0.0",
+      mkdirp: "^2.1.2",
+      prettier: "^2.5.1",
+      rimraf: "^5.0.0",
+      "source-map-support": "^0.5.9",
+      typescript: "~5.2.0",
+      "@rollup/plugin-commonjs": "^24.0.0",
+      "@rollup/plugin-json": "^6.0.0",
+      "@rollup/plugin-multi-entry": "^6.0.0",
+      "@rollup/plugin-node-resolve": "^13.1.3",
+      rollup: "^2.66.1",
+      "rollup-plugin-sourcemaps": "^0.6.3",
+      "uglify-js": "^3.4.9"
+    }
+  } as any;
+  if (codeModel.clients.length > 1) {
+    delete packageInfo.exports["./api"];
+    delete packageInfo.exports["./models"];
+    for (const client of codeModel.clients) {
+      const subfolder = normalizeName(
+        client.name.replace("Client", ""),
+        NameType.File
+      );
+      packageInfo.exports[`./${subfolder}`] = {
+        types: `./types/src/${subfolder}/index.d.ts`,
+        import: `./dist-esm/src/${subfolder}/index.js`
+      };
+      packageInfo.exports[`./${subfolder}/api`] = {
+        types: `./types/src/${subfolder}/api/index.d.ts`,
+        import: `./dist-esm/src/${subfolder}/api/index.js`
+      };
+      packageInfo.exports[`./${subfolder}/models`] = {
+        types: `./types/src/${subfolder}/models/index.d.ts`,
+        import: `./dist-esm/src/${subfolder}/models/index.js`
+      };
+    }
+  }
+  if (codeModel.options.hierarchyClient) {
+    for (const client of codeModel.clients) {
+      for (const operationGroup of client.operationGroups) {
+        if (operationGroup.namespaceHierarchies.length === 0) {
+          continue;
+        }
+        const subfolder =
+          codeModel.clients.length > 1
+            ? normalizeName(client.name.replace("Client", ""), NameType.File)
+            : undefined;
+        const subApiPath = `api/${getClassicalLayerPrefix(
+          operationGroup,
+          NameType.File,
+          "/"
+        )}`;
+        packageInfo.exports[
+          `./${subfolder ? subfolder + "/" : ""}${subApiPath}`
+        ] = {
+          types: `./types/src/${
+            subfolder ? subfolder + "/" : ""
+          }${subApiPath}/index.d.ts`,
+          import: `./dist-esm/src/${
+            subfolder ? subfolder + "/" : ""
+          }${subApiPath}/index.js`
+        };
+      }
+    }
+  }
+  return packageInfo;
```

Comment:
it would be great if we can encapsulate this logic and share it between the branded and unbranded codepaths.

Reply:
updated